src/pudl/output/ferc1.py

```diff
@@ -422,6 +445,53 @@ def out_ferc1__yearly_steam_plants_fuel_sched402(
     return fuel_df
 
 
+## example/draft factory pattern
+specs = [
+    {"name": "ferc1__yearly_purchased_power_and_exchanges_sched326"},
+    {"name": "ferc1__yearly_plant_in_service_sched204"},
+    {"name": "ferc1__yearly_balance_sheet_assets_sched110"},
+]
+
+
+# draft asset_factory
+def generate_asset_factory(spec) -> AssetsDefinition:
+    var_name = "core_" + spec["name"]
+    core_ = globals()[var_name]
+
+    @asset(
+        name=f"_out_{get_core_ferc1_asset_description(spec["name"])}",
+        io_manager_key="pudl_io_manager",
+        compute_kind="Python",
+    )
+    def _asset(
+        core_: pd.DataFrame,
+        core_pudl__assn_ferc1_pudl_utilities: pd.DataFrame,
+    ) -> pd.DataFrame:
+        """Generate a dataframe for {} asset specification.""".format(spec["name"])
+        return_df = core_.merge(
+            core_pudl__assn_ferc1_pudl_utilities, on="utility_id_ferc1"
+        )
+        return return_df
+
+    return _asset
+
+
+# defs = Definition(assets=[generate_asset_factory(spec) for spec in specs])
+
+
+def create_generated_assets() -> list[AssetsDefinition]:
+    """Create a list of generated FERC Form 1 assets.
+
+    Returns:
+        A list of :class:`AssetsDefinitions` where each asset is an generated FERC Form 1
+        table.
+    """
+    return [generate_asset_factory(**kwargs) for kwargs in specs]
+
+
+exploded_ferc1_assets = create_generated_assets()
```

Comment:
I don't think a whole separate function is needed here, we can generate the `AssetsDefinitions` for dagster to pick up on module import with something like:

```py
ferc1_output_assets = [
    ferc1_output_asset_factory(table_name) for table name in
    [
        "yearly_purchased_power_and_exchanges_sched326",
        "yearly_plant_in_service_sched204",
        "yearly_balance_sheet_assets_sched110",
    ]
]
```